dev/tools/localization/gen_l10n_types.dart

```diff
@@ -423,6 +423,23 @@ class AppResourceBundleCollection {
         languageToLocales[bundle.locale.languageCode].add(bundle.locale);
       }
     }
+
+    languageToLocales.forEach((String language, List<LocaleInfo> listOfCorrespondingLocales) {
+      final List<String> localeStrings = listOfCorrespondingLocales.map((LocaleInfo locale) {
+        return locale.toString();
+      }).toList();
+      if (!localeStrings.contains(language)) {
+        throw L10nException(
+          'Arb file for a fallback, $language, does not exist, even though \n'
+          'the following locale(s) exist: $listOfCorrespondingLocales. \n'
+          'When locales specify a script code or country code, a \n'
+          'base locale (without the script code or country code) should \n'
+          'exist as the fallback. Please create a {fileName}_$language.arb \n'
```

Comment:
Was {fileName} suppose to refer to a variable?

Reply:
No, it is meant to be a placeholder value for whatever the arb file name's prefix was. Is there a better way to communicate that?